libs/application/core/src/lib/ApplicationTemplateHelper.ts

```diff
@@ -187,4 +189,41 @@ export class ApplicationTemplateHelper<
     }
     return roleInState.write
   }
+
+  async applyAnswerValidators(
+    newAnswers: FormValue,
+  ): Promise<undefined | Record<string, string>> {
+    const validators = this.template.answerValidators
+
+    if (!validators) {
+      return Promise.resolve(undefined)
+    }
+
+    let hasError = false
+    const errorMap: Record<string, string> = {}
+    const validatorPaths = Object.keys(validators)
+
+    for (let i = 0; i < validatorPaths.length; i++) {
+      const validatorPath = validatorPaths[i]
```

Comment:
Minor: since you're not using the index for anything else you could also use `for (const validatorPath of validatorPaths)`